core/trino-main/src/main/java/io/trino/operator/DeleteAndInsertMergeProcessor.java

```diff
@@ -129,12 +133,12 @@ public Page transformPage(Page inputPage)
             long operation = TINYINT.getLong(operationChannelBlock, position);
             if (operation != DEFAULT_CASE_OPERATION_NUMBER) {
                 // Delete and Update because both create a delete row
-                if (operation == DELETE_OPERATION_NUMBER || operation == UPDATE_OPERATION_NUMBER) {
-                    addDeleteRow(pageBuilder, inputPage, position);
+                if (operation == DELETE_OPERATION_NUMBER || operation == UPDATE_OPERATION_NUMBER || operation == UPDATE_DELETE_OPERATION_NUMBER) {
```

Comment:
Did we just verify above that `operation == UPDATE_DELETE_OPERATION_NUMBER` is impossible?